stdlib/src/collections/index_normalization.mojo

```diff
@@ -0,0 +1,81 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+"""The utilities provided in this module help normalize the access 
+to data elements in arrays."""
+
+
+fn _get_out_of_bounds_error_message[
```

Comment:
Unfortunately, this will allocate heap memory, which is not allowed on certain platforms. This might be one reason why the `__refitem__` below uses a `debug_assert`.

Reply:
Answered above :)